contrib/babelfishpg_tsql/src/pl_funcs-2.c

```diff
@@ -350,6 +351,67 @@ coalesce_typmod_hook_impl(const CoalesceExpr *cexpr)
 	return exprTypmod((Node *) linitial(cexpr->args));
 }
 
+void
+check_restricted_stored_procedure_hook_impl(Oid proc_id)
+{
+	/*
+	 * BABEL-4390
+	 * List of procedure names that are not allowed to be dropped. These procedures 
+	 * are considered essential or restricted due to security or operational reasons.
+	 */
+	const char		*restricted_procedures[] = {
+		"xp_qv",
+		"xp_instance_regread",
+		"sp_addlinkedsrvlogin",
+		"sp_droplinkedsrvlogin",
+		"sp_dropserver",
+		"sp_enum_oledb_providers",
+		"sp_testlinkedserver"
+	};
+	const int		RESTRICTED_PROCEDURES_COUNT = sizeof(restricted_procedures) / sizeof(restricted_procedures[0]);
+	HeapTuple		proctup;
+	Form_pg_proc		procform;
+	const char		*procname;
+	Oid		schema_oid = InvalidOid;
+	Oid		sys_oid = InvalidOid;
+	Oid		dbo_oid = InvalidOid;
+	bool		is_restricted = false;
+
+	proctup = SearchSysCache1(PROCOID, ObjectIdGetDatum(proc_id));
+	if (!HeapTupleIsValid(proctup))
+	{
+		elog(ERROR, "cache lookup failed for procedure %u", proc_id);
+		return;
+	}
+
+	procform = (Form_pg_proc) GETSTRUCT(proctup);
+	procname = pstrdup(NameStr(procform->proname));
+	schema_oid = procform->pronamespace;
+	sys_oid = get_namespace_oid("sys", true);
```

Comment:
Question: Should this have `missing_ok = false` , with true set the function can return an InvalidOid and we are not checking if its valid in next logic